### PR TITLE
✨ Set balance to u128 instead of config type on pallet_funding

### PR DESCRIPTION
**Pull request by JuaniRios**
## What?
- Remove the `Balance` config type, and replace it for u128

## Why?
- Having to access our fundamental balance type through trait interfaces was making our code unnecessarily bloated. The only reason to have a generic balance type is if our pallet is written to be used in many other blockchains, which is not the case.
- Now we can use directly u128, instead of having to convert with into()
- We were already assuming 99% that it was u128 since it had to implement From<u128>
 
## How?
- Replace BalanceOf<T> with Balance
- Delete all unnecessary conversion
- Remove the generic Balance types from the pallet funding types

## Testing?
Normal tests

## Anything Else?
This will now allow us to use abs_diff for the vesting api

### CODE REVIEW

**Comment by JuaniRios** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/Polimec/polimec-node/394?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#394** <a href="https://app.graphite.dev/github/pr/Polimec/polimec-node/394?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#393** <a href="https://app.graphite.dev/github/pr/Polimec/polimec-node/393?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#392** <a href="https://app.graphite.dev/github/pr/Polimec/polimec-node/392?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#391** <a href="https://app.graphite.dev/github/pr/Polimec/polimec-node/391?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#390** <a href="https://app.graphite.dev/github/pr/Polimec/polimec-node/390?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#389** <a href="https://app.graphite.dev/github/pr/Polimec/polimec-node/389?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#386** <a href="https://app.graphite.dev/github/pr/Polimec/polimec-node/386?utm_source=stack-comment-icon" target="_blank">Graphite</a>: 1 other dependent PR ([#387](https://github.com/Polimec/polimec-node/pull/387) <a href="https://app.graphite.dev/github/pr/Polimec/polimec-node/387?utm_source=stack-comment-icon" target="_blank">Graphite</a>)
* **#383** <a href="https://app.graphite.dev/github/pr/Polimec/polimec-node/383?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#384** <a href="https://app.graphite.dev/github/pr/Polimec/polimec-node/384?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#385** <a href="https://app.graphite.dev/github/pr/Polimec/polimec-node/385?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#388** <a href="https://app.graphite.dev/github/pr/Polimec/polimec-node/388?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @JuaniRios and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

**Review by lrazovic — APPROVED** [review]
I see the simplification this brings in terms of syntax, but I would wait for feedback from @vstam1 before merging this

File: integration-tests/src/tests/e2e.rs
Comment (JuaniRios) [reply integration-tests/src/tests/e2e.rs:514]: true, do you have that number? I can add it.

Comment (JuaniRios) [reply integration-tests/src/tests/e2e.rs:514]: Actually, at this part of the code, that snippet was unnecessary, so I removed it. We check that the CT is migrated correctly some lines below.

**Review by vstam1 — APPROVED** [review]
Lgtm

**Review by vstam1 — APPROVED** [review]
Lgtm

**Comment by JuaniRios** [comment]
### Merge activity

* **Sep 12, 9:39 AM EDT**: @JuaniRios started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/Polimec/polimec-node/394).

**Pull request merged**
